card/mod/notifications/set/type/email_template/email_config.rb

```diff
@@ -1,61 +1,81 @@
-def email_config args={}
-  config = {}
-  args[:context] ||= self
+EMAIL_FIELDS =
+  %i[to from cc bcc attach subject text_message html_message].freeze
 
-  %i[to from cc bcc].each do |field_name|
-    process_email_field(field_name, config, args) do |field_card|
-      field_card.process_email_addresses(args[:context], format: "email_text")
-    end
-  end
-  process_email_field(:attach, config, args) do |field_card|
-    field_card.extended_item_contents args[:context]
-  end
-  process_message_field :subject, config, args, "email_text",
-                        content_opts: { chunk_list: :nest_only }
-  process_message_field :text_message, config, args, "email_text"
-
-  from_name, from_email =
-    if config[:from] =~ /(.*)\<(.*)>/
-      [Regexp.last_match(1).strip, Regexp.last_match(2)]
-    else
-      [nil, config[:from]]
-    end
+EMAIL_FIELD_METHODS =
+  { subject: :contextual_content,
+    text_message: :contextual_content,
+    attach: :extended_item_contents }.freeze
 
-  if (default_from = Card::Mailer.default[:from])
-    config[:from] =
-      if from_email
-        %("#{from_name || from_email}" <#{default_from}>)
-      else
-        default_from
-      end
-    config[:reply_to] ||= config[:from]
-  elsif config[:from].blank?
-    config[:from] = Card[Card::WagnBotID].account.email
+# @param [Card] context  the card in whose context all email fields will be interpreted
+# @param [Hash] fields override any templated field configurations with hash values
+# @param [Hash] opts options for rendering. unknown options become format options
+# @option opts [Card, String, Integer] :auth user identifier. render as this user
+def email_config context, fields={}, opts={}
```

Comment:
this was actually the hardest api change.  all opts other than :auth are treated as format opts, which allows us to set instance variables in the format classes created.

Hmm, need to add that comment...